File: backend/node-express/src/db/seed.ts
import Database from 'better-sqlite3';
import fs from 'fs';
import path from 'path';

const DB_PATH = path.resolve(__dirname, '../../worldcup.db');
const SEED_PATH = path.resolve(__dirname, '../../../../seed-data/matches.json');

console.log('🌱 Seeding database...');
console.log(`   Database: ${DB_PATH}`);
console.log(`   Seed data: ${SEED_PATH}`);

// Read seed data
const rawData = fs.readFileSync(SEED_PATH, 'utf-8');
const seedData = JSON.parse(rawData);

// Create/open database
const db = new Database(DB_PATH);
db.pragma('journal_mode = WAL');
db.pragma('foreign_keys = ON');

// Read and execute schema
const schemaPath = path.resolve(__dirname, './schema.sql');
const schema = fs.readFileSync(schemaPath, 'utf-8');

// Drop existing tables
db.exec('DROP TABLE IF EXISTS flight_prices');
db.exec('DROP TABLE IF EXISTS itinerary_stops');
db.exec('DROP TABLE IF EXISTS itineraries');
db.exec('DROP TABLE IF EXISTS matches');
db.exec('DROP TABLE IF EXISTS teams');
db.exec('DROP TABLE IF EXISTS cities');

// Create tables
db.exec(schema);

// Insert cities
const insertCity = db.prepare(
  'INSERT INTO cities (id, name, country, latitude, longitude, stadium, accommodation_per_night) VALUES (?, ?, ?, ?, ?, ?, ?)'
);
for (const city of seedData.cities) {
  insertCity.run(city.id, city.name, city.country, city.latitude, city.longitude, city.stadium, city.accommodationPerNight || 150);
}
console.log(`   ✅ Inserted ${seedData.cities.length} cities`);

// Insert teams
const insertTeam = db.prepare(
  'INSERT INTO teams (id, name, code, group_name) VALUES (?, ?, ?, ?)'
);
for (const team of seedData.teams) {
  insertTeam.run(team.id, team.name, team.code, team.group);
}
console.log(`   ✅ Inserted ${seedData.teams.length} teams`);

// Insert matches
const insertMatch = db.prepare(
  'INSERT INTO matches (id, home_team_id, away_team_id, city_id, kickoff, group_name, match_day, ticket_price) VALUES (?, ?, ?, ?, ?, ?, ?, ?)'
);
for (const match of seedData.matches) {
  insertMatch.run(
    match.id,
    match.homeTeamId,
    match.awayTeamId,
    match.cityId,
    match.kickoff,
    match.group,
    match.matchDay,
    match.ticketPrice || 100
  );
}
console.log(`   ✅ Inserted ${seedData.matches.length} matches`);

// Insert flight prices
if (seedData.flightPrices) {
  const insertFlight = db.prepare(
    'INSERT INTO flight_prices (id, origin_city_id, destination_city_id, price_usd) VALUES (?, ?, ?, ?)'
  );
  for (const flight of seedData.flightPrices) {
    insertFlight.run(flight.id, flight.originCityId, flight.destinationCityId, flight.priceUsd);
  }
  console.log(`   ✅ Inserted ${seedData.flightPrices.length} flight prices`);
}

db.close();
console.log('🎉 Database seeded successfully!');

File: backend/python-flask/app/seed.py
"""
Database seeding script.

Usage: python -m app.seed
"""

import json
import os
import sys

# Add parent directory to path so we can import app
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app import create_app
from app.db import db
from app.models.city import City
from app.models.team import Team
from app.models.match import Match
from app.models.flight_price import FlightPrice

SEED_PATH = os.path.join(os.path.dirname(__file__), '..', '..', '..', 'seed-data', 'matches.json')


def seed():
    app = create_app()

    with app.app_context():
        print('🌱 Seeding database...')
        print(f'   Seed data: {os.path.abspath(SEED_PATH)}')

        # Read seed data
        with open(SEED_PATH, 'r') as f:
            data = json.load(f)

        # Drop and recreate all tables
        db.drop_all()
        db.create_all()

        # Insert cities
        for city_data in data['cities']:
            city = City(
                id=city_data['id'],
                name=city_data['name'],
                country=city_data['country'],
                latitude=city_data['latitude'],
                longitude=city_data['longitude'],
                stadium=city_data['stadium'],
                accommodation_per_night=city_data.get('accommodationPerNight', 150.0),
            )
            db.session.add(city)
        print(f'   ✅ Inserted {len(data["cities"])} cities')

        # Insert teams
        for team_data in data['teams']:
            team = Team(
                id=team_data['id'],
                name=team_data['name'],
                code=team_data['code'],
                group_name=team_data['group'],
            )
            db.session.add(team)
        print(f'   ✅ Inserted {len(data["teams"])} teams')

        # Insert matches
        for match_data in data['matches']:
            match = Match(
                id=match_data['id'],
                home_team_id=match_data['homeTeamId'],
                away_team_id=match_data['awayTeamId'],
                city_id=match_data['cityId'],
                kickoff=match_data['kickoff'],
                group_name=match_data['group'],
                match_day=match_data['matchDay'],
                ticket_price=match_data.get('ticketPrice', 100.0),
            )
            db.session.add(match)
        print(f'   ✅ Inserted {len(data["matches"])} matches')

        # Insert flight prices
        if 'flightPrices' in data:
            for flight_data in data['flightPrices']:
                flight = FlightPrice(
                    id=flight_data['id'],
                    origin_city_id=flight_data['originCityId'],
                    destination_city_id=flight_data['destinationCityId'],
                    price_usd=flight_data['priceUsd'],
                )
                db.session.add(flight)
            print(f'   ✅ Inserted {len(data["flightPrices"])} flight prices')

        db.session.commit()
        print('🎉 Database seeded successfully!')


if __name__ == '__main__':
    seed()
